Lx 224 & Trinity Mods

knickels opened the pull request:
Following up on a couple of good suggestions from #50

Roger-random commented on modifications/kevin_nickels/README.md:
```diff
@@ -12,20 +20,26 @@ The [LX-16A](https://www.amazon.com/Hiwonder-LX-16A-Robotic-Controller-Control/d
 
 ## Joystick Controller Modifications
-I also modified the [Handheld Arduino Joystick controller](https://cad.onshape.com/documents/47ee84411a76e86eae57ea0f/w/f70ae835a2d548c4b95fa46e/e/c8538a52492e953cdcdd3db5?renderMode=0&uiState=6887d5ec43a024454e4cecf6) to accept an [RJ9](https://www.ebay.com/itm/317094616170) so that I can connect my controller to my rover with a standard coiled phone cord.
+I also modified the [Handheld Arduino Joystick controller](https://cad.onshape.com/documents/47ee84411a76e86eae57ea0f/w/f70ae835a2d548c4b95fa46e/e/c8538a52492e953cdcdd3db5?renderMode=0&uiState=688a32537f05d177d7ca5baf) to accept an [RJ9](https://www.ebay.com/itm/317094616170) so that I can connect my controller to my rover with a standard coiled phone cord.
```
I've been fiddling with these OnShape links and my current hypothesis is that the suffix "?renderMode=0&uiState=[...]" is somehow specific to your browser. If I use the full URL I get an error but if I remove that suffix I could access your OnShape CAD.

Please truncate URL by removing "?" and everything after that. (I arbitrarily chose this OnShape URL to comment upon but the request applies to all of them on this page.)

knickels replied:
I don't think that I had "Anyone with link" set to "Can View" at first - I assumed that when you copied a link using the chain icon at the top that it set the permissions properly.  

I just tested all four of the OnShape links in an incognito window and in a chrome, edge, and firefox windows with no OnShape cookie, and they all worked fine for me.  Would you mind checking again?  BTW, what browser are you using?

Roger-random replied:
Same trio: WIndows Chrome, Firefox, and Edge. But it worked today so maybe it was a temporary glitch yesterday on OnShape's end. Shrug. Looks good to go now.

Roger-random replied:
... and it's giving me 403 Forbidden again. I don't know what variables changed between the earlier attempt and my most recent attempt. I can't very well demand you to fix something I can't articulate. And since it does work occasionally (if for reason's I don't understand) I'll to merge this now.

![Trippy Onshape CAD 403 Forbidden](https://github.com/user-attachments/assets/62889fd4-c4f8-44f8-a51d-e98c25c177ba)

knickels commented:
Hm, I thought that the sharing permissions had fixed it.  I want it to work
for folks...
Can you look at the versions without the renderMode and uiState arguments
in
https://github.com/NickelsLab/Sawppy_Rover/tree/lx-224/modifications/kevin_nickels,
and I'll make a new PR if that works for you?
Sorry this has been so hard :)
--kn

On Thu, Jul 31, 2025 at 1:36 PM Roger Cheng ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In modifications/kevin_nickels/README.md
> <https://github.com/Roger-random/Sawppy_Rover/pull/51#discussion_r2246109974>
> :
>
> >
>
>  ## Joystick Controller Modifications
> -I also modified the [Handheld Arduino Joystick controller](https://cad.onshape.com/documents/47ee84411a76e86eae57ea0f/w/f70ae835a2d548c4b95fa46e/e/c8538a52492e953cdcdd3db5?renderMode=0&uiState=6887d5ec43a024454e4cecf6) to accept an [RJ9](https://www.ebay.com/itm/317094616170) so that I can connect my controller to my rover with a standard coiled phone cord.
> +I also modified the [Handheld Arduino Joystick controller](https://cad.onshape.com/documents/47ee84411a76e86eae57ea0f/w/f70ae835a2d548c4b95fa46e/e/c8538a52492e953cdcdd3db5?renderMode=0&uiState=688a32537f05d177d7ca5baf) to accept an [RJ9](https://www.ebay.com/itm/317094616170) so that I can connect my controller to my rover with a standard coiled phone cord.
>
> ... and it's giving me 403 Forbidden again. I don't know what variables
> changed between the earlier attempt and my most recent attempt. I can't
> very well demand you to fix something I can't articulate. And since it does
> work occasionally (if for reason's I don't understand) I'll to merge this
> now.
>
> Trippy.Onshape.CAD.403.Forbidden.jpg (view on web)
> <https://github.com/user-attachments/assets/62889fd4-c4f8-44f8-a51d-e98c25c177ba>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Roger-random/Sawppy_Rover/pull/51#discussion_r2246109974>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDZENVYKFMJ7ZOAQE2GEOD3LJO2BAVCNFSM6AAAAACCXKKHUWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTANZWGUYDKNZQGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kevin Nickels ***@***.***> 210-999-7543
https://www.trinity.edu/directory/knickels
Spring 2025 Schedule/Student Hours
<https://docs.google.com/spreadsheets/d/1CjTLrXfn9C8Ky4xsZas95mxJ5glWGlkU5EOM-Z9E9Io/edit?usp=sharing>
Associate Professor, Engineering Science, Trinity University
Center for Sciences and Innovation (CSI), Rm 470Q, San Antonio TX 78212-7200

Roger-random commented:
I noticed that when the full link works as expected, Onshape itself trims the rendermode and uistate arguments. See address bar in screen shot:

![onshape itself trims suffix in url](https://github.com/user-attachments/assets/c9b68f26-b92a-433a-9d22-2ab976dc4756)

Based on what I see so far, I'm inclined to believe this is an Onshape issue. Their link sharing feature should not have included uistate and rendermode in its generated URL. It doesn't seem to do anything useful and risk resulting in 403 Forbidden for reasons opaque to me. (Either that, or the URL parser should gracefully handle unexpected arguments. Or maybe both!)

I don't think you need to apologize for Onshape goof, but I will be happy to merge a PR with those arguments removed.

Given my sad history of interaction with Onshape customer support, I do not plan to spend any time filing an issue about this. You are welcome to do so if you so choose.
